### ISSUE TITLE
Wrong behavior for omitNothingFields = True 

### ISSUE THREAD

**Issue by gip** [bos/aeson#454]
Using Aeson `0.11.2.0`. See the following code: 
```haskell
{-# LANGUAGE DeriveGeneric #-}
{-# LANGUAGE OverloadedStrings #-}
module Test where
import GHC.Generics
import Data.Aeson
import Data.Aeson.Types

data A = A {
  a :: Int,
  b :: Maybe Int
} deriving (Show, Generic)
instance ToJSON A where
  toEncoding = genericToEncoding $ defaultOptions { omitNothingFields = True }

test = encode $ A 0 Nothing
```

`test` is an invalid JSON string: `{"a":0,}`

**Comment by bergmark** [comment]
This might have been fixed already, can you please try this on the master branch?

**Comment by bergmark** [comment]
I tested it, result on master is `"{\"a\":0}"` so this has been fixed.

**Comment by gip** [comment]
Can we do a fix for `0.11.2.x`?

**Comment by bergmark** [comment]
I think this was fixed by #396 which doesn't contain a breaking change so it should be possible, can you send a PR?

**Comment by bergmark** [comment]
PR should be against the 0.11.x branch by the way

**Comment by yogsototh** [comment]
+1 should be fixed in the 0.11 branch.

**Comment by bergmark** [comment]
I'm pretty busy with a new job at the moment, but if someone can fix this on the 0.11.x branch I'll be happy to make a release.
